packages/data/src/components/use-select/test/index.js

```diff
@@ -132,5 +132,89 @@ describe( 'useSelect', () => {
 			children: 'bar',
 		} );
 	} );
+	describe( 'rerenders as expected with various mapSelect return ' +
+		'types', () => {
+		const getComponent = ( mapSelectSpy, dependencyKey ) => ( props ) => {
+			const dependencies = props[ dependencyKey ];
+			mapSelectSpy.mockImplementation(
+				( select ) => select( 'testStore' ).testSelector( props.keyName )
+			);
+			const data = useSelect( mapSelectSpy, [ dependencies ] );
+			return <div>{ data }</div>;
+		};
+		let selectorSpy,
+			subscribedSpy,
+			subscriberSpy,
+			selectSpy,
+			TestComponent;
+		beforeEach( () => {
+			selectorSpy = jest.fn();
+			subscriberSpy = jest.fn().mockImplementation(
+				( subscription ) => {
+					subscribedSpy = subscription;
+				}
+			);
+			registry.registerStore( 'testStore', {
+				reducer: () => null,
+				selectors: {
+					testSelector: selectorSpy,
+				},
+			} );
+			registry.subscribe = subscriberSpy;
+			selectSpy = jest.fn();
+			TestComponent = jest.fn().mockImplementation(
+				getComponent( selectSpy, 'keyName' )
+			);
+		} );
+		[
+			[
+				'boolean',
+				[ false, true ],
+			],
+			[
+				'number',
+				[ 10, 20 ],
+			],
```

Comment:
We should also test objects and null <=> undefined.

Reply:
Added additional tests in de7f4fc (modification to one in 46f80f6) for covering null/undefined/objects.  Had to change the test component so tests would validate on prop rather than children (because objects cannot be returned as the value for children).